File: code_01_01.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
std::ifstream input_file("./Inputs/input_01.txt");

int sum = 0;
int first_digit = -1;
int last_digit = -1;

if(input_file.is_open()){
    char c;
    while(input_file){
        c = input_file.get();
        if(c == '\n'){
            sum += first_digit*10 + last_digit;
            first_digit = -1;
            last_digit = -1;
        }
        else if(isdigit(c)){
            last_digit = c - '0';
            if(first_digit == -1){
                first_digit = c - '0';
            }
        }
    }
}

std::cout << sum;

}

File: code_01_01.py
input_file = './Inputs/input_01.txt'

with open(input_file) as f:
    dat = f.read()

sum = 0
first_digit = None
last_digit = None

for c in dat:
    if(c == '\n'):
        sum += first_digit*10 + last_digit
        first_digit = None
        last_digit = None
    elif(c.isdigit()):
        last_digit = int(c)
        if(first_digit is None):
            first_digit = int(c)

print(sum)